beacon_node/beacon_chain/src/block_verification_types.rs

```diff
@@ -162,10 +163,32 @@ impl<E: EthSpec> RpcBlock<E> {
     ) -> Result<Self, AvailabilityCheckError> {
         let block_root = block_root.unwrap_or_else(|| get_block_root(&block));
 
-        if block.num_expected_blobs() > 0 && custody_columns.is_empty() {
-            // The number of required custody columns is out of scope here.
-            return Err(AvailabilityCheckError::MissingCustodyColumns);
+        let column_commitments = custody_columns
+            .iter()
+            .map(|c| c.as_data_column().kzg_commitments.clone())
+            .unique()
+            .at_most_one()
+            .map_err(|_| AvailabilityCheckError::KzgCommitmentsInconsistent)?;
+
+        if let (Some(column_commitments), Ok(block_commitments)) = (
+            column_commitments,
+            block.message().body().blob_kzg_commitments(),
+        ) {
+            if column_commitments.len() != block_commitments.len() {
+                return Err(AvailabilityCheckError::KzgCommitmentListMismatch);
+            }
+            for (&column_commitment, &block_commitment) in
+                column_commitments.iter().zip(block_commitments.iter())
+            {
+                if column_commitment != block_commitment {
```

Comment:
I actually want to remove this check for blob sidecars since the inclusion proof should ensure that this never happens. Have you observed this inconsistency issue on devnets?

Reply:
Yeah I actually ran into a few KZG errors during sync. 

I think you raised a good point and this PR shouldn't be required. The proper fix is to ensure that we verify the inclusion proof - I think we're missing this in sampling and by range rpc. I'll raise an issue.

Reply:
Raised issue #6111. Closing this one